packages/core/src/supertokens.ts

```diff
@@ -100,8 +100,8 @@ export const parsePublicDataToken = (token: string) => {
   }
 }
 
-export const useSession = () => {
-  const [publicData, setPublicData] = useState(publicDataStore.emptyPublicData)
+export const useSession = (initialPublicData?: PublicData) => {
+  const [publicData, setPublicData] = useState(initialPublicData ?? publicDataStore.emptyPublicData)
   const [isLoading, setIsLoading] = useState(true)
```

Comment:
I think if we pass the publicData as props we should set the isLoadingState to false and not run the `useIsomorphicLayoutEffect` at all.

Reply:
Good call, just made these changes